src/api/chassis/model/hDriveModel.cpp

```diff
@@ -0,0 +1,247 @@
+    } else {
+      leftOutput = maxInput;
+      rightOutput = forwardSpeed - yaw;
+    }
+  }
+
+  leftSideMotor->moveVoltage(static_cast<int16_t>(std::clamp(leftOutput, -1.0, 1.0) * maxVoltage));
+  rightSideMotor->moveVoltage(
+    static_cast<int16_t>(std::clamp(rightOutput, -1.0, 1.0) * maxVoltage));
+}
+
+void HDriveModel::hArcade(const double ixSpeed,
+                          const double iforwardSpeed,
+                          const double iyaw,
+                          const double ithreshold) {
+  double xSpeed = std::clamp(ixSpeed, -1.0, 1.0);
+  if (std::abs(xSpeed) < ithreshold) {
+    xSpeed = 0;
+  }
+
+  double forwardSpeed = std::clamp(iforwardSpeed, -1.0, 1.0);
+  if (std::abs(forwardSpeed) < ithreshold) {
+    forwardSpeed = 0;
+  }
+
+  double yaw = std::clamp(iyaw, -1.0, 1.0);
+  if (std::abs(yaw) < ithreshold) {
+    yaw = 0;
+  }
+
+  leftSideMotor->moveVoltage(
+    static_cast<int16_t>(std::clamp(forwardSpeed + yaw, -1.0, 1.0) * maxVoltage));
+  rightSideMotor->moveVoltage(
+    static_cast<int16_t>(std::clamp(forwardSpeed - yaw, -1.0, 1.0) * maxVoltage));
+  middleMotor->moveVoltage(static_cast<int16_t>(std::clamp(xSpeed, -1.0, 1.0) * maxVoltage));
+}
+
+void HDriveModel::left(const double ispeed) {
+  leftSideMotor->moveVelocity(static_cast<int16_t>(std::clamp(ispeed, -1.0, 1.0) * maxVelocity));
+}
+
+void HDriveModel::right(const double ispeed) {
+  rightSideMotor->moveVelocity(static_cast<int16_t>(std::clamp(ispeed, -1.0, 1.0) * maxVelocity));
+}
+
+void HDriveModel::middle(const double ispeed) {
+  middleMotor->moveVelocity(static_cast<int16_t>(std::clamp(ispeed, -1.0, 1.0) * maxVelocity));
+}
+
+std::valarray<std::int32_t> HDriveModel::getSensorVals() const {
+  return std::valarray<std::int32_t>{static_cast<std::int32_t>(leftSensor->get()),
+                                     static_cast<std::int32_t>(rightSensor->get()),
+                                     static_cast<std::int32_t>(middleSensor->get())};
+}
+
+void HDriveModel::resetSensors() {
+  leftSensor->reset();
+  rightSensor->reset();
+  middleSensor->reset();
+}
+
+void HDriveModel::setBrakeMode(const AbstractMotor::brakeMode mode) {
+  leftSideMotor->setBrakeMode(mode);
+  rightSideMotor->setBrakeMode(mode);
+  middleMotor->setBrakeMode(mode);
+}
+
+void HDriveModel::setEncoderUnits(const AbstractMotor::encoderUnits units) {
+  leftSideMotor->setEncoderUnits(units);
+  rightSideMotor->setEncoderUnits(units);
+  middleMotor->setEncoderUnits(units);
+}
+
+void HDriveModel::setGearing(const AbstractMotor::gearset gearset) {
+  leftSideMotor->setGearing(gearset);
+  rightSideMotor->setGearing(gearset);
+  middleMotor->setGearing(gearset);
+}
+
+void HDriveModel::setMaxVelocity(double imaxVelocity) {
+  if (imaxVelocity < 0) {
+    maxVelocity = 0;
+  } else {
+    maxVelocity = imaxVelocity;
+  }
+}
+
+double HDriveModel::getMaxVelocity() const {
+  return maxVelocity;
+}
+
+void HDriveModel::setMaxVoltage(double imaxVoltage) {
```

Comment:
I'm not sure if this is the same thing for all the models, but this function can be simplified to
```cpp
maxVoltage = std::clamp(imaxVoltage, 0, 12000);